packages/presentation/lib/screens/movie_details/movie_details_bloc.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:get_it/get_it.dart';
+import 'package:domain/usecases/get_image_url_usecase.dart';
+import 'package:domain/usecases/get_movie_cast_usecase.dart';
+import 'package:presentation/bloc/base/bloc.dart';
+import 'package:presentation/bloc/base/bloc_impl.dart';
+import 'package:presentation/const.dart';
+import 'package:presentation/screens/movie_details/data/movie_details_data.dart';
+import 'package:presentation/screens/movie_details/movie_details_screen.dart';
+
+abstract class MovieDetailsBloc implements Bloc<MovieDetailsData> {
+  factory MovieDetailsBloc(
+    GetImageUrlUsecase getImageUrlUsecase,
+    GetMovieCastUsecase getMovieCastUsecase,
+  ) =>
+      _MovieDetailsBloc(
+        getImageUrlUsecase,
+        getMovieCastUsecase,
+      );
+
+  String? getImageUrlById(String? id);
+
+  Future<void> getCast(int movieId);
+
+  void handleBackPressed();
+}
+
+class _MovieDetailsBloc extends BlocImpl<MovieDetailsData>
+    implements MovieDetailsBloc {
+  final GetImageUrlUsecase _getImageUrlUsecase;
+  final GetMovieCastUsecase _getMovieCastUsecase;
+
+  _MovieDetailsBloc(
+    this._getImageUrlUsecase,
+    this._getMovieCastUsecase,
+  ) : super(
+          initState: MovieDetailsData.init(),
+        );
+
+  @override
+  void initState() {
+    _updateMovieDetailsWithArgumentsData();
+  }
+
+  @override
+  String? getImageUrlById(String? id) {
+    return id != null ? _getImageUrlUsecase(id) : id;
+  }
+
+  @override
+  Future<void> getCast(int movieId) async {
+    final params = GetMovieCastUsecaseParams(
+        movieId, MovieDetailsScreenConfig.maxCastCount);
+
+    final cast = await _getMovieCastUsecase(params);
+
+    final newState = MovieDetailsData(state.movieDetails, cast);
+
+    add(newState);
+  }
+
+  @override
+  void handleBackPressed() {
+    appNavigator.pop();
+  }
+
+  Future<void> _updateMovieDetailsWithArgumentsData() async {
```

Comment:
u have separate callback in BaseBloc - initArgs. this will allow us to partially avoid this method

Reply:
@sergei-mikhailovskii-idf Unfortunately cannot find such method (initArgs) in my project(

Reply:
check the implementation in @ekaterina-demeshko-idf project please